libraries/AP_Terrain/tools/create_terrain.py

```diff
@@ -0,0 +1,299 @@
+        self.lon = ref_lon
+
+    def fill(self, gx, gy, altitude):
+        '''fill a square'''
+        self.height[gx][gy] = int(altitude)
+
+    def blocknum(self):
+        '''find IO block number'''
+        stride = east_blocks(self.lat_degrees*1e7, self.lon_degrees*1e7)
+        return stride * self.grid_idx_x + self.grid_idx_y
+
+class DataFile(object):
+    def __init__(self, lat, lon):
+        if lat < 0:
+            NS = 'S'
+        else:
+            NS = 'N'
+        if lon < 0:
+            EW = 'W'
+        else:
+            EW = 'E'
+        name = "terrain/%c%02u%c%03u.DAT" % (NS, min(abs(int(lat)), 99),
+                                        EW, min(abs(int(lon)), 999))
+        try:
+            os.mkdir("terrain")
+        except Exception:
+            pass
+        if not os.path.exists(name):
+            self.fh = open(name, 'w+b')
+        else:
+            self.fh = open(name, 'r+b')
+
+    def seek_offset(self, block):
+        '''seek to right offset'''
+        # work out how many longitude blocks there are at this latitude
+        file_offset = block.blocknum() * IO_BLOCK_SIZE
+        self.fh.seek(file_offset)
+
+    def pack(self, block):
+        '''pack into a block'''
+        buf = bytes()
+        buf += struct.pack("<QiiHHH", block.bitmap, block.lat, block.lon, block.crc, block.version, block.spacing)
+        for gx in range(TERRAIN_GRID_BLOCK_SIZE_X):
+            buf += struct.pack("<%uh" % TERRAIN_GRID_BLOCK_SIZE_Y, *block.height[gx])
+        buf += struct.pack("<HHhb", block.grid_idx_x, block.grid_idx_y, block.lon_degrees, block.lat_degrees)
+        return buf
+
+    def write(self, block):
+        '''write a grid block'''
+        self.seek_offset(block)
+        block.crc = 0
+        buf = self.pack(block)
+        block.crc = crc16.crc16xmodem(buf)
+        buf = self.pack(block)
+        self.fh.write(buf)
+
+    def check_filled(self, block):
+        '''read a grid block and check if already filled'''
+        self.seek_offset(block)
+        buf = self.fh.read(IO_BLOCK_SIZE)
+        if len(buf) != IO_BLOCK_SIZE:
+            return False
+        (bitmap, lat, lon, crc, version, spacing) = struct.unpack("<QiiHHH", buf[:22])
+        if (version != TERRAIN_GRID_FORMAT_VERSION or
+            abs(lat - block.lat)>2 or
+            abs(lon - block.lon)>2 or
+            spacing != GRID_SPACING or
+            bitmap != (1<<56)-1):
+            return False
+        buf = buf[:16] + struct.pack("<H", 0) + buf[18:]
+        crc2 = crc16.crc16xmodem(buf[:1821])
+        if crc2 != crc:
+            return False
+        return True
+
+def create_degree(lat, lon):
+    '''create data file for one degree lat/lon'''
+    lat_int = int(math.floor(lat))
+    lon_int = int(math.floor((lon)))
+
+    tiles = {}
+
+    dfile = DataFile(lat_int, lon_int)
+
+    print("Creating for %d %d" % (lat_int, lon_int))
+
+    total_blocks = east_blocks(lat_int*1e7, lon_int*1e7) * TERRAIN_GRID_BLOCK_SIZE_Y
```

Comment:
Should be `int(math.ceil(112000 / (TERRAIN_GRID_BLOCK_SPACING_X * float(GRID_SPACING))))` instead of TERRAIN_GRID_BLOCK_SIZE_Y